### PR TITLE
Upgrade to WiX4. Boostrapper1 and MetaMorpheusSetup

### PR DESCRIPTION
**Pull request by elaboy**
#### Using the FireGiant WiXV4 Visual Studio 2022 extension, right-click on the project and select `Upgrade to WiX V4...`. Click `Okay` when the pop-up appears to convert the file format.

image

#### After upgrade, the project would not build, referring to an error at the `boostrapper1.wxs` source file, indicating problems with `DisplayInternalUI`. The fix was gathered from https://github.com/wixtoolset/issues/issues/7170

`<MsiPackage bal:DisplayInternalUICondition="yes" SourceFile="$(var.MetaMorpheusSetup.TargetPath)" />`

## All code was upgraded to WiXV4, unlike last PR where only the boostrapper was upgraded.

### CODE REVIEW

File: MetaMorpheusSetup/MetaMorpheusSetup.wixproj [hunk 1]
```diff
@@ -1,31 +1,22 @@
...
   <PropertyGroup>
-    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
     <Platform Condition=" '$(Platform)' == '' ">x64</Platform>
-    <ProductVersion>3.10</ProductVersion>
-    <ProjectGuid>be72541a-66a4-4958-9d19-56acc7d3876b</ProjectGuid>
-    <SchemaVersion>2.0</SchemaVersion>
     <OutputName>MetaMorpheusInstaller</OutputName>
-    <OutputType>Package</OutputType>
     <Name>MetaMorpheusSetup</Name>
+    <OutputPath>bin\$(Configuration)\</OutputPath>
```
Comment (elaboy) [inline MetaMorpheusSetup/MetaMorpheusSetup.wixproj:7]: FireGiant's convertion extension deleted this line, changing the msi output path and ultimately throwing an error on appveyor due not finding the msi.

File: Bootstrapper1/Bootstrapper.wixproj [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<Project Sdk="WixToolset.Sdk/4.0.0-rc.2">
```
Comment (acesnik) [inline Bootstrapper1/Bootstrapper.wixproj:1]: Please change the name of the folder from `Bootstrapper1` to `Bootstrapper` as well

Comment (acesnik) [reply Bootstrapper1/Bootstrapper.wixproj:1]: Or you could rename it to `MetaMorpheusBootstrapper` which would group it with the other wix project.

**Review by acesnik — COMMENTED** [review]
I'm still not getting success here... not sure what the issue is, but figured I'd post the build messages I'm getting.

With Bootstrapper:
```
Build started...
------ Build started: Project: Bootstrapper1, Configuration: Debug x86 ------
		C:\Program Files (x86)\WiX Toolset v3.11\bin\candle.exe -dDebug -d"DevEnvDir=C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\\" -dSolutionDir=C:\Users\User\source\repos\MetaMorpheus\ -dSolutionExt=.sln -dSolutionFileName=MetaMorpheus.sln -dSolutionName=MetaMorpheus -dSolutionPath=C:\Users\User\source\repos\MetaMorpheus\MetaMorpheus.sln -dConfiguration=Debug -dOutDir=bin\Debug\ -dPlatform=x86 -dProjectDir=C:\Users\User\source\repos\MetaMorpheus\Bootstrapper1\ -dProjectExt=.wixproj -dProjectFileName=Bootstrapper1.wixproj -dProjectName=Bootstrapper1 -dProjectPath=C:\Users\User\source\repos\MetaMorpheus\Bootstrapper1\Bootstrapper1.wixproj -dTargetDir=C:\Users\User\source\repos\MetaMorpheus\Bootstrapper1\bin\Debug\ -dTargetExt=.exe -dTargetFileName=MetaMorpheusInstaller.exe -dTargetName=MetaMorpheusInstaller -dTargetPath=C:\Users\User\source\repos\MetaMorpheus\Bootstrapper1\bin\Debug\MetaMorpheusInstaller.exe -dMetaMorpheusSetup.Configuration=Debug -d"MetaMorpheusSetup.FullConfiguration=Debug|x64" -dMetaMorpheusSetup.Platform=x64 -dMetaMorpheusSetup.ProjectDir=C:\Users\User\source\repos\MetaMorpheus\MetaMorpheusSetup\ -dMetaMorpheusSetup.ProjectExt=.wixproj -dMetaMorpheusSetup.ProjectFileName=MetaMorpheusSetup.wixproj -dMetaMorpheusSetup.ProjectName=MetaMorpheusSetup -dMetaMorpheusSetup.ProjectPath=C:\Users\User\source\repos\MetaMorpheus\MetaMorpheusSetup\MetaMorpheusSetup.wixproj -dMetaMorpheusSetup.TargetDir=C:\Users\User\source\repos\MetaMorpheus\MetaMorpheusSetup\bin\x64\Debug\ -dMetaMorpheusSetup.TargetExt=.msi -dMetaMorpheusSetup.TargetFileName=MetaMorpheusInstaller.msi -dMetaMorpheusSetup.TargetName=MetaMorpheusInstaller -dMetaMorpheusSetup.TargetPath=C:\Users\User\source\repos\MetaMorpheus\MetaMorpheusSetup\bin\x64\Debug\MetaMorpheusInstaller.msi -out obj\Debug\ -arch x86 -ext "C:\Program Files (x86)\WiX Toolset v3.11\bin\\WixUtilExtension.dll" -ext "C:\Program Files (x86)\WiX Toolset v3.11\bin\\WixBalExtension.dll" bootstrapper1.wxs bootstrapper.wxproj
C:\Users\User\source\repos\MetaMorpheus\Bootstrapper1\bootstrapper1.wxs(1,0): error CNDL0199: The Wix element has an incorrect namespace of 'http://wixtoolset.org/schemas/v4/wxs'.  Please make the Wix element look like the following: <Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">.
C:\Users\User\source\repos\MetaMorpheus\Bootstrapper1\bootstrapper.wxproj(1,0): error CNDL0048: The document element name 'Project' is invalid.  A Windows Installer XML source file must use 'Wix' as the document element name.
Done building project "Bootstrapper1.wixproj" -- FAILED.

========== Build: 0 succeeded or up-to-date, 1 failed, 0 skipped ==========
========== Elapsed 00:00,526 ==========
```

With MetaMorpheusSetup:
```
Build started...
1>------ Build started: Project: MetaMorpheusSetup, Configuration: Debug x64 ------
1>C:\Users\User\.nuget\packages\wixtoolset.sdk\4.0.0-rc.1\tools\wix.targets(218,5): error : The C:\Users\User\source\repos\MetaMorpheus\MetaMorpheusSetup\obj\project.assets.json file does not exist. Restore the project before trying to build again. Restore and build with "MSBuild -Restore" or "dotnet build".
========== Build: 0 succeeded, 1 failed, 5 up-to-date, 0 skipped ==========
========== Elapsed 00:00,268 ==========
```

File: Bootstrapper1/Bootstrapper.wixproj (hunk 2 above)
Comment (elaboy) [reply Bootstrapper1/Bootstrapper.wixproj:1]: I'll try grouping both, appreciate the feedback. :)

**Comment by elaboy** [comment]
> I'm still not getting success here... not sure what the issue is, but figured I'd post the build messages I'm getting.
> 
> With Bootstrapper:
> 
> ```
> Build started...
> ------ Build started: Project: Bootstrapper1, Configuration: Debug x86 ------
> 		C:\Program Files (x86)\WiX Toolset v3.11\bin\candle.exe -dDebug -d"DevEnvDir=C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\\" -dSolutionDir=C:\Users\User\source\repos\MetaMorpheus\ -dSolutionExt=.sln -dSolutionFileName=MetaMorpheus.sln -dSolutionName=MetaMorpheus -dSolutionPath=C:\Users\User\source\repos\MetaMorpheus\MetaMorpheus.sln -dConfiguration=Debug -dOutDir=bin\Debug\ -dPlatform=x86 -dProjectDir=C:\Users\User\source\repos\MetaMorpheus\Bootstrapper1\ -dProjectExt=.wixproj -dProjectFileName=Bootstrapper1.wixproj -dProjectName=Bootstrapper1 -dProjectPath=C:\Users\User\source\repos\MetaMorpheus\Bootstrapper1\Bootstrapper1.wixproj -dTargetDir=C:\Users\User\source\repos\MetaMorpheus\Bootstrapper1\bin\Debug\ -dTargetExt=.exe -dTargetFileName=MetaMorpheusInstaller.exe -dTargetName=MetaMorpheusInstaller -dTargetPath=C:\Users\User\source\repos\MetaMorpheus\Bootstrapper1\bin\Debug\MetaMorpheusInstaller.exe -dMetaMorpheusSetup.Configuration=Debug -d"MetaMorpheusSetup.FullConfiguration=Debug|x64" -dMetaMorpheusSetup.Platform=x64 -dMetaMorpheusSetup.ProjectDir=C:\Users\User\source\repos\MetaMorpheus\MetaMorpheusSetup\ -dMetaMorpheusSetup.ProjectExt=.wixproj -dMetaMorpheusSetup.ProjectFileName=MetaMorpheusSetup.wixproj -dMetaMorpheusSetup.ProjectName=MetaMorpheusSetup -dMetaMorpheusSetup.ProjectPath=C:\Users\User\source\repos\MetaMorpheus\MetaMorpheusSetup\MetaMorpheusSetup.wixproj -dMetaMorpheusSetup.TargetDir=C:\Users\User\source\repos\MetaMorpheus\MetaMorpheusSetup\bin\x64\Debug\ -dMetaMorpheusSetup.TargetExt=.msi -dMetaMorpheusSetup.TargetFileName=MetaMorpheusInstaller.msi -dMetaMorpheusSetup.TargetName=MetaMorpheusInstaller -dMetaMorpheusSetup.TargetPath=C:\Users\User\source\repos\MetaMorpheus\MetaMorpheusSetup\bin\x64\Debug\MetaMorpheusInstaller.msi -out obj\Debug\ -arch x86 -ext "C:\Program Files (x86)\WiX Toolset v3.11\bin\\WixUtilExtension.dll" -ext "C:\Program Files (x86)\WiX Toolset v3.11\bin\\WixBalExtension.dll" bootstrapper1.wxs bootstrapper.wxproj
> C:\Users\User\source\repos\MetaMorpheus\Bootstrapper1\bootstrapper1.wxs(1,0): error CNDL0199: The Wix element has an incorrect namespace of 'http://wixtoolset.org/schemas/v4/wxs'.  Please make the Wix element look like the following: <Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">.
> C:\Users\User\source\repos\MetaMorpheus\Bootstrapper1\bootstrapper.wxproj(1,0): error CNDL0048: The document element name 'Project' is invalid.  A Windows Installer XML source file must use 'Wix' as the document element name.
> Done building project "Bootstrapper1.wixproj" -- FAILED.
> 
> ========== Build: 0 succeeded or up-to-date, 1 failed, 0 skipped ==========
> ========== Elapsed 00:00,526 ==========
> ```
> 
> With MetaMorpheusSetup:
> 
> ```
> Build started...
> 1>------ Build started: Project: MetaMorpheusSetup, Configuration: Debug x64 ------
> 1>C:\Users\User\.nuget\packages\wixtoolset.sdk\4.0.0-rc.1\tools\wix.targets(218,5): error : The C:\Users\User\source\repos\MetaMorpheus\MetaMorpheusSetup\obj\project.assets.json file does not exist. Restore the project before trying to build again. Restore and build with "MSBuild -Restore" or "dotnet build".
> ========== Build: 0 succeeded, 1 failed, 5 up-to-date, 0 skipped ==========
> ========== Elapsed 00:00,268 ==========
> ```

Interesting, I am not sure why it is building using WiXv3... I have no problems building in my PC. Anyone else experiencing same thing?

Thanks for the feedback @acesnik

**Comment by acesnik** [comment]
Oh, I see what's going on. I was on the branch from your other PR.

Here's what I'm seeing on this branch:

Whole Build:
```
Build started...
1>------ Skipped Build: Project: Bootstrapper, Configuration: Debug x86 ------
1>Project not selected to build for this solution configuration 
Restored C:\Users\User\source\repos\MetaMorpheus\EngineLayer\EngineLayer.csproj (in 79 ms).
Restored C:\Users\User\source\repos\MetaMorpheus\TaskLayer\TaskLayer.csproj (in 81 ms).
Restored C:\Users\User\source\repos\MetaMorpheus\CMD\CMD.csproj (in 8 ms).
Restored C:\Users\User\source\repos\MetaMorpheus\GuiFunctions\GuiFunctions.csproj (in 128 ms).
Restored C:\Users\User\source\repos\MetaMorpheus\Test\Test.csproj (in 126 ms).
2>------ Build started: Project: EngineLayer, Configuration: Debug Any CPU ------
Restored C:\Users\User\source\repos\MetaMorpheus\GUI\GUI.csproj (in 50 ms).
2>C:\Users\User\source\repos\MetaMorpheus\EngineLayer\Ms2ScanWithSpecificMass.cs(62,53,63,115): warning CS0618: 'MzSpectrum.Deconvolute(MzRange, int, int, double, double)' is obsolete: 'Deconvolution Has been moved to the Deconvoluter Object'
2>EngineLayer -> C:\Users\User\source\repos\MetaMorpheus\EngineLayer\bin\Debug\net6.0\EngineLayer.dll
2>Done building project "EngineLayer.csproj".
3>------ Build started: Project: GuiFunctions, Configuration: Debug Any CPU ------
4>------ Build started: Project: TaskLayer, Configuration: Debug Any CPU ------
3>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "C:\Users\User\.nuget\packages\mzlib\1.0.537\lib\net6.0-windows7.0\BayesianEstimation.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project.
3>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "C:\Users\User\.nuget\packages\mzlib\1.0.537\lib\net6.0-windows7.0\Chemistry.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project.
3>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "C:\Users\User\source\repos\MetaMorpheus\EngineLayer\bin\Debug\net6.0\EngineLayer.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project.
3>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "C:\Users\User\.nuget\packages\mzlib\1.0.537\lib\net6.0-windows7.0\FlashLFQ.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project.
3>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "C:\Users\User\.nuget\packages\mzlib\1.0.537\lib\net6.0-windows7.0\MassSpectrometry.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project.
3>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "C:\Users\User\.nuget\packages\mzlib\1.0.537\lib\net6.0-windows7.0\Mgf.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project.
3>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "C:\Users\User\.nuget\packages\mzlib\1.0.537\lib\net6.0-windows7.0\MzIdentML.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project.
3>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "C:\Users\User\.nuget\packages\mzlib\1.0.537\lib\net6.0-windows7.0\MzLibSpectralAveraging.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project.
3>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "C:\Users\User\.nuget\packages\mzlib\1.0.537\lib\net6.0-windows7.0\MzLibUtil.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project.
3>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "C:\Users\User\.nuget\packages\mzlib\1.0.537\lib\net6.0-windows7.0\MzML.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project.
3>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "C:\Users\User\.nuget\packages\mzlib\1.0.537\lib\net6.0-windows7.0\mzPlot.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project.
3>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "C:\Users\User\.nuget\packages\mzlib\1.0.537\lib\net6.0-windows7.0\PepXML.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project.
3>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "C:\Users\User\.nuget\packages\mzlib\1.0.537\lib\net6.0-windows7.0\Proteomics.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project.
3>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "C:\Users\User\.nuget\packages\spectralaveraging\1.0.6\lib\net6.0-windows7.0\SpectralAveraging.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project.
3>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "C:\Users\User\.nuget\packages\mzlib\1.0.537\lib\net6.0-windows7.0\ThermoRawFileReader.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project.
3>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "C:\Users\User\.nuget\packages\mzlib\1.0.537\lib\net6.0-windows7.0\UsefulProteomicsDatabases.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project.
3>GuiFunctions -> C:\Users\User\source\repos\MetaMorpheus\GuiFunctions\bin\Debug\net6.0-windows\GuiFunctions.dll
3>Done building project "GuiFunctions.csproj".
4>C:\Users\User\source\repos\MetaMorpheus\TaskLayer\MetaMorpheusTask.cs(145,71,149,82): warning CS0612: 'MsDataScan.GetIsolatedMassesAndCharges(MzSpectrum, int, int, double, double)' is obsolete
4>TaskLayer -> C:\Users\User\source\repos\MetaMorpheus\TaskLayer\bin\Debug\net6.0\TaskLayer.dll
4>Done building project "TaskLayer.csproj".
5>------ Build started: Project: Test, Configuration: Debug Any CPU ------
6>------ Build started: Project: GUI, Configuration: Debug Any CPU ------
7>------ Build started: Project: CMD, Configuration: Debug Any CPU ------
6>GUI -> C:\Users\User\source\repos\MetaMorpheus\GUI\bin\Debug\net6.0-windows\MetaMorpheusGUI.dll
7>CMD -> C:\Users\User\source\repos\MetaMorpheus\CMD\bin\Debug\net6.0\CMD.dll
8>------ Build started: Project: MetaMorpheusSetup, Configuration: Debug x64 ------
8>C:\Users\User\.nuget\packages\wixtoolset.sdk\4.0.0-rc.2\tools\wix.targets(233,5): error : The C:\Users\User\source\repos\MetaMorpheus\MetaMorpheusSetup\obj\project.assets.json file does not exist. Restore the project before trying to build again. Restore and build with "MSBuild -Restore" or "dotnet build".
8>Done building project "MetaMorpheusSetup.wixproj" -- FAILED.
8>
5>Test -> C:\Users\User\source\repos\MetaMorpheus\Test\bin\Debug\net6.0-windows\Test.dll
========== Build: 6 succeeded, 1 failed, 0 up-to-date, 1 skipped ==========
========== Elapsed 00:12,977 ==========
```

**Comment by elaboy** [comment]
## I edited your post by mistake, I apologize... @acesnik 

### I have no problems with my PC, I'll try it with another one in the morning. Thanks!

image

**Comment by Alexander-Sol** [comment]
It appears that Appveyor doesn't have Wix v4 installed on the build image. It may be using wix v3? I found what appears to be an error message that may be caused by this
```
MSBuild version 17.4.0+18d5aef85 for .NET Framework
MSBUILD : error MSB1009: Project file does not exist.
Switch: C:\projects\metamorpheus\Bootstrapper1\Bootstrapper1.wixproj
Zipping Command-Line Version
```

**Comment by Alexander-Sol** [comment]
image

I see this whenever I start a newly installed version of MetaMorpheus. May not have been caused by this PR, but maybe something to look into

**Comment by acesnik** [comment]
Might be time to switch from AppVeyor to Github Actions 😉

**Comment by acesnik** [comment]
https://github.com/smith-chem-wisc/MetaMorpheus/pull/2128

**Comment by elaboy** [comment]
image

https://github.com/smith-chem-wisc/MetaMorpheus/pull/2224/commits/4144b2eb3177ca18612b1d611eb4cd95f9dd53d3 Fixes the `EXCELDEFAULTPROPERTY` warning.

**Comment by elaboy** [comment]
I have a theory on what could be happening, will work on that this week.

**Comment by elaboy** [comment]
> image
> 
> I see this whenever I start a newly installed version of MetaMorpheus. May not have been caused by this PR, but maybe something to look into. MM runs fine after the warning is closed

I have a theory on what could be causing this warning. Will work on it this week.

**Comment by acesnik** [comment]
Looks like you changed the folder to `Bootstrapper`. Did you want to change it to `MetaMorpheusBootstrapper` to group them alphabetically in the folder?

**Comment by Alexander-Sol** [comment]
> > image
> > I see this whenever I start a newly installed version of MetaMorpheus. May not have been caused by this PR, but maybe something to look into. MM runs fine after the warning is closed
> 
> I have a theory on what could be causing this warning. Will work on it this week.
> 
> EDIT: @nbollis and I fixed the warning pop up. [4bcf1fe](https://github.com/smith-chem-wisc/MetaMorpheus/commit/4bcf1fe8cdc61d0dacd9df2def6f969ef4e02a3b)

I figured out how to fix the error on startup. In MetaUpdater.xaml.cs, line 125, replace the current line with the following
`releases.NavigateToString(allVersionsText.Length == 0 ? " " : allVersionsText.ToString());`

**Review by avcarr2 — APPROVED** [review]

**Review by Alexander-Sol — APPROVED** [review]
Looking good. MSBuild errors are no longer generated in Appveyor. The .msi installs, and the error message no longer appears on start up. I think we're ready for release

**Review by trishorts — APPROVED** [review]

**Comment by acesnik** [comment]
It's so nice to see just `Bootstrapper`; thanks for changing that as part of this! 😅

**Pull request merged**